diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae4c8464..29de4a75 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,7 @@ project(cesium-native
 option(PRIVATE_CESIUM_SQLITE "ON to rename SQLite symbols to cesium_sqlite3_* so they won't conflict with other SQLite implemenentations" OFF)
 option(CESIUM_TRACING_ENABLED "Whether to enable the Cesium performance tracing framework (CESIUM_TRACE_* macros)." OFF)
 option(CESIUM_COVERAGE_ENABLED "Whether to enable code coverage" OFF)
-option(CESIUM_TESTS_ENABLED "Whether to enable tests" ON)
+option(CESIUM_TESTS_ENABLED "Whether to enable tests" OFF)
 
 if (CESIUM_TRACING_ENABLED)
     add_compile_definitions(CESIUM_TRACING_ENABLED=1)
@@ -84,7 +84,7 @@ function(configure_cesium_library targetName)
     if (MSVC)
         target_compile_options(${targetName} PRIVATE /W4 /WX /wd4201 /bigobj)
     else()
-        target_compile_options(${targetName} PRIVATE -Werror -Wall -Wextra -Wconversion -Wpedantic -Wshadow -Wsign-conversion)
+        target_compile_options(${targetName} PRIVATE -Wall -Wextra -Wconversion -Wpedantic -Wshadow -Wsign-conversion -matomics -mbulk-memory -pthread)
     endif()
 
     set_target_properties(${targetName} PROPERTIES
diff --git a/CesiumUtility/CMakeLists.txt b/CesiumUtility/CMakeLists.txt
index 8b566b8c..4366c9a9 100644
--- a/CesiumUtility/CMakeLists.txt
+++ b/CesiumUtility/CMakeLists.txt
@@ -5,14 +5,14 @@ configure_cesium_library(CesiumUtility)
 cesium_glob_files(CESIUM_UTILITY_SOURCES src/*.cpp)
 cesium_glob_files(CESIUM_UTILITY_HEADERS src/*.h)
 cesium_glob_files(CESIUM_UTILITY_PUBLIC_HEADERS include/CesiumUtility/*.h)
-cesium_glob_files(CESIUM_UTILITY_TEST_SOURCES test/*.cpp)
-cesium_glob_files(CESIUM_UTILITY_TEST_HEADERS test/*.h)
-
-set_target_properties(CesiumUtility
-    PROPERTIES
-        TEST_SOURCES "${CESIUM_UTILITY_TEST_SOURCES}"
-        TEST_HEADERS "${CESIUM_UTILITY_TEST_HEADERS}"
-)
+#cesium_glob_files(CESIUM_UTILITY_TEST_SOURCES test/*.cpp)
+#cesium_glob_files(CESIUM_UTILITY_TEST_HEADERS test/*.h)
+
+#set_target_properties(CesiumUtility
+#    PROPERTIES
+#        TEST_SOURCES "${CESIUM_UTILITY_TEST_SOURCES}"
+#        TEST_HEADERS "${CESIUM_UTILITY_TEST_HEADERS}"
+#)
 
 set_target_properties(CesiumUtility
     PROPERTIES
diff --git a/extern/CMakeLists.txt b/extern/CMakeLists.txt
index fafca408..edb4b89f 100644
--- a/extern/CMakeLists.txt
+++ b/extern/CMakeLists.txt
@@ -101,6 +101,8 @@ target_include_directories(s2geometry SYSTEM PUBLIC s2geometry/src)
 if (MSVC)
   target_compile_options(s2geometry PRIVATE "/J" "/wd4127" "/wd4996")
   target_compile_definitions(s2geometry PRIVATE NOMINMAX _USE_MATH_DEFINES)
+else()
+  target_compile_options(s2geometry PRIVATE -matomics -mbulk-memory)
 endif()
 
 set(HTTPLIB_USE_BROTLI_IF_AVAILABLE OFF CACHE BOOL "Don't use Brotli")
diff --git a/extern/sqlite3/CMakeLists.txt b/extern/sqlite3/CMakeLists.txt
index 58fc2b5f..5102b61f 100644
--- a/extern/sqlite3/CMakeLists.txt
+++ b/extern/sqlite3/CMakeLists.txt
@@ -10,6 +10,8 @@ target_compile_definitions(sqlite3 PUBLIC SQLITE_THREADSAFE=1)
 file(READ ${CMAKE_CURRENT_SOURCE_DIR}/sqlite3.h SQLITE3_HEADER_CONTENTS)
 file(READ ${CMAKE_CURRENT_SOURCE_DIR}/sqlite3.c SQLITE3_SOURCE_CONTENTS)
 
+target_compile_options(sqlite3 PRIVATE -fno-exceptions -matomics -mbulk-memory -pthread)
+
 if (PRIVATE_CESIUM_SQLITE)
     string(REPLACE sqlite3 cesium_sqlite3 SQLITE3_HEADER_CONTENTS "${SQLITE3_HEADER_CONTENTS}")
     string(REPLACE sqlite3 cesium_sqlite3 SQLITE3_SOURCE_CONTENTS "${SQLITE3_SOURCE_CONTENTS}")
